Libraries/LibWeb/HTML/StructuredSerialize.cpp

```diff
@@ -1209,6 +1247,9 @@ static WebIDL::ExceptionOr<GC::Ref<Bindings::PlatformObject>> create_transferred
         TRY(message_port->transfer_receiving_steps(transfer_data_holder));
         return message_port;
     }
+    default:
```

Comment:
Not having a default here was intentional, the compiler will warn if you forgot an enum value. The enum value should be already validated from the wire by the time we call this one

Reply:
Great point, thanks :)

`default` case is added since `ArrayBuffer`s are not platform objects, hence cannot be created by this function.

However, I agree that it is much better to have the warning from the compiler. I replaced `default` with 2 specific cases for both `ArrayBuffer` and `ResizableArrayBuffer`. Hopefully, it makes the intent more explicit.

From what I see, it might be a good idea to split `TransferType` enum into 2 smaller enums, e.g. `InterfaceTransferType` (with `MessagePort` and other interfaces/platform object) and `BufferTransferType`. Might be a bad idea, though ;)

Also, there is a note that says (line `1015`):
```
// FIXME: Consolidate this function with the similar is_interface_exposed_on_target_realm() used when transferring objects.
//        Also, the name parameter would be better off being the interface name (as a string) so that we don't need a switch statement.
```